spec/support/webmock.rb

```diff
@@ -2,18 +2,33 @@
 [:delete, :get, :post, :put, :patch].each do |method|
   self.class.send(:define_method, "a_#{method}") do |path, *args|
     opts = args.last.is_a?(Hash) ? args.pop : {}
+
     ver = args.first || opts[:api_version] || 1
     fmt = opts[:api_format] || :json
-    a_request(method, Gemfury.endpoint + path).with(
-      :headers => {
-        'Accept' => "application/vnd.fury.v#{ver}+#{fmt}",
-        'X-Gem-Version' => Gemfury::VERSION
-      }
-    )
+    http_accept = 'application/vnd.fury.v%s+%s' % [ ver, fmt ]
+    if opts.include?(:http_accept)
+      http_accept = '%s, %s' % [ opts[:http_accept], http_accept ]
+    end
+
+    endpoint = opts[:endpoint] || Gemfury.endpoint
+
+    with_opts = {
+      :headers => { 'Accept' => http_accept,
+                    'X-Gem-Version' => Gemfury::VERSION }
+    }
+
+    if opts.include?(:query)
+      with_opts[:query] = opts[:query]
+    end
+
+    a_request(method, endpoint + (path || '')).with(with_opts)
```

Comment:
I don't think we need a default path here.  It may hide benign errors when a path is unintentionally `nil`.  So I'd rather be explicit at the call site with `''` in this use case.